pkg/kapp/config/conf.go

```diff
@@ -178,3 +178,21 @@ func (c Conf) ChangeRuleBindings() []ChangeRuleBinding {
 	}
 	return result
 }
+
+type StripNameHashSuffixConfigs []StripNameHashSuffixConfig
+
+func (c StripNameHashSuffixConfigs) AggregateToCtlRes() [][]ctlres.ResourceMatcher {
```

Comment:
What do you think about adding a `ResourceMatcher` method to `config.go`? Something like [this](https://github.com/vmware-tanzu/carvel-kapp/blob/develop/pkg/kapp/config/config.go#L312) (this also uses the AnyMatcher which I hope would reduce some additional complexity in `stripnamehassuffix.go`)

I am assuming that we would want users to be able to use something like below to be able to strip the suffix for configmaps and secrets.
```
stripNameHashSuffix:
- resourceMatchers:
  - apiVersionKindMatcher: {apiVersion: v1, kind: ConfigMap}
  - apiVersionKindMatcher: {apiVersion: v1, kind: Secret}
```